### PR TITLE
Fix shard coverage

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

File: pgt/test/Test.hs [hunk 1]
```diff
@@ -49,6 +53,39 @@ setupSchema pgConfig = do
...
+  , test "one shard, one element"             shardConfigOne     ["A"]      ["A"]
+  , test "one shard, two elements"            shardConfigOne     ["A", "B"] ["A", "B"]
+  , test "two shards, index 0, empty input"   (shardConfigTwo 0) []         []
+  , test "two shards, index 1, empty input"   (shardConfigTwo 1) []         []
+  , test "two shards, index 0, one element"   (shardConfigTwo 0) []         ["A"]
+  , test "two shards, index 1, one element"   (shardConfigTwo 1) ["A"]      ["A"]
+  , test "two shards, index 0, two elements"  (shardConfigTwo 0) ["A"]      ["A", "B"]
+  , test "two shards, index 1, two elements"  (shardConfigTwo 1) ["B"]      ["A", "B"]
+  , test "two shards, index 0, tree elements" (shardConfigTwo 0) ["A"]      ["A", "B", "C"]
+  , test "two shards, index 1, tree elements" (shardConfigTwo 1) ["B", "C"] ["A", "B", "C"]
```
Comment (snusnu) [inline pgt/test/Test.hs:68]: @mbj `s/tree/three/`

File: pgt/src/PGT.hs [hunk 2]
```diff
@@ -66,21 +66,24 @@ parseShardConfig count@(ShardCount countValue) index@(ShardIndex indexValue) =
...
-selectShard :: ShardConfig -> Vector Test -> Vector Test
-selectShard ShardConfig{count=(ShardCount count), index=(ShardIndex index)} tests =
-    Vector.slice (convertImpure startIndex) (convertImpure effectiveChunkSize) tests
+selectShard :: ShardConfig -> Vector a -> Vector a
+selectShard ShardConfig{count=(ShardCount count), index=(ShardIndex index)} items =
+    Vector.slice (convertImpure startIndex) (convertImpure effectiveChunkSize) items
   where
     nominalChunkSize :: Natural
-    nominalChunkSize = convertImpure (Vector.length tests) `div` count
+    nominalChunkSize = length `div` count
```
Comment (dkubb) [inline pgt/src/PGT.hs:74]: __question:__ I don't suppose it's possible for `count` to be `0`?

Comment (mbj) [reply pgt/src/PGT.hs:74]: yes, see the constructor of `ShardCount`.

Comment (mbj) [reply pgt/src/PGT.hs:74]: well `parseShardCount` you'll not find the constructor exported.

Comment (snusnu) [reply pgt/src/PGT.hs:74]: @dkubb `parseShardCount` forbids that.

**Pull request merged**